### DISCUSSION TITLE
add multi output file path for cli generated interfaces

### DISCUSSION THREAD

**Discussion by armatrix** [wevm/wagmi#3959] [Ideas] [answered]
Project may have multi contracts, the generated files should be separated according to the contract to make maintenance easier.
Are the following two methods feasible?

- Modify the configuration of out to mapping or array to support multiple files.
- Add out filed to the contract configuration to support specified files.
sth like this:

`export default defineConfig({
  out: 'src/generated.ts',  //  1. need array or mapping like for multiple outputs
  contracts: [
    {
      name: 'foo',
      abi: foo.abi,
      // output: '', //  2. optional, output file for generated
    },
    {
      name: 'bar',
      abi: bar.abi,
    },
    ...
  ],
  plugins: [],
})
`

**Comment by tmm** [comment, marked as answer]
Check out the [Array Config docs](https://wagmi.sh/cli/config/configuring-cli#array-config).
